jetty-slf4j-impl/src/main/java/org/eclipse/jetty/logging/JettyLoggerFactory.java

```diff
@@ -19,15 +19,28 @@
 import java.util.concurrent.ConcurrentMap;
 import java.util.function.Consumer;
 import java.util.function.Function;
+import javax.management.Attribute;
+import javax.management.AttributeList;
+import javax.management.AttributeNotFoundException;
+import javax.management.DynamicMBean;
+import javax.management.MBeanAttributeInfo;
+import javax.management.MBeanConstructorInfo;
+import javax.management.MBeanException;
+import javax.management.MBeanInfo;
+import javax.management.MBeanNotificationInfo;
+import javax.management.MBeanOperationInfo;
+import javax.management.MBeanParameterInfo;
+import javax.management.ReflectionException;
 
 import org.slf4j.ILoggerFactory;
 import org.slf4j.Logger;
 
-public class JettyLoggerFactory implements ILoggerFactory, JettyLoggerFactoryMBean
+public class JettyLoggerFactory implements ILoggerFactory, DynamicMBean, JettyLoggerFactoryMBean
 {
     private final JettyLoggerConfiguration configuration;
     private final JettyLogger rootLogger;
     private final ConcurrentMap<String, JettyLogger> loggerMap;
+    private MBeanInfo mBeanInfo;
```

Comment:
I would make this `final` and build it in the constructor (calling a method).
This would solve the problem of `getMBeanInfo()` possibly creating it twice when accessed concurrently.

Reply:
So we create this MBeanInfo always? even for those users that don't actually use JMX?

Reply:
Good point!